sqlmesh/schedulers/airflow/common.py

```diff
@@ -52,6 +52,9 @@ class PlanApplicationRequest(PydanticModel):
     models_to_backfill: t.Optional[t.Set[str]]
     end_bounded: bool
     ensure_finalized_snapshots: bool
+    directly_modified_snapshots: t.List[SnapshotId]
+    indirectly_modified_snapshots: t.Dict[SnapshotId, t.Set[SnapshotId]]
```

Comment:
This key is likely not going to work with pydantic. AFAIK, Pydantic doesn't support (de)serialization of complex types as dict keys / set items.